### PR TITLE
Register ClusterOperator to properly generate event for the kind

### PR DESCRIPTION
**Pull request by ingvagabund**
```
E0401 08:06:27.851998       1 event.go:259] Could not construct reference to: '&v1.ClusterOperator{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"machine-api", GenerateName:"", Namespace:"", SelfLink:"/apis/config.openshift.io/v1/clusteroperators/machine-api", UID:"0cfdf4cd-5455-11e9-a124-0adb1443898a", ResourceVersion:"656", Generation:1, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:63689702787, loc:(*time.Location)(0x1d1cea0)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:""}, Spec:v1.ClusterOperatorSpec{}, Status:v1.ClusterOperatorStatus{Conditions:[]v1.ClusterOperatorStatusCondition(nil), Versions:[]v1.OperandVersion(nil), RelatedObjects:[]v1.ObjectReference(nil), Extension:runtime.RawExtension{Raw:[]uint8(nil), Object:runtime.Object(nil)}}}' due to: 'no kind is registered for the type v1.ClusterOperator in scheme "github.com/openshift/cluster-api/pkg/client/clientset_generated/clientset/scheme/register.go:31"'. Will not report event: 'Normal' 'Status upgrade' 'Progressing towards operator: 0.0.1-snapshot'
E0401 08:13:09.777804       1 event.go:259] Could not construct reference to: '&v1.ClusterOperator{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"machine-api", GenerateName:"", Namespace:"", SelfLink:"/apis/config.openshift.io/v1/clusteroperators/machine-api", UID:"0cfdf4cd-5455-11e9-a124-0adb1443898a", ResourceVersion:"2856", Generation:1, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:63689702787, loc:(*time.Location)(0x1d1cea0)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:""}, Spec:v1.ClusterOperatorSpec{}, Status:v1.ClusterOperatorStatus{Conditions:[]v1.ClusterOperatorStatusCondition{v1.ClusterOperatorStatusCondition{Type:"Progressing", Status:"False", LastTransitionTime:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, Reason:"SyncingResources", Message:"Running resync for operator: 0.0.1-snapshot"}, v1.ClusterOperatorStatusCondition{Type:"Available", Status:"True", LastTransitionTime:v1.Time{Time:time.Time{wall:0x0, ext:63689702787, loc:(*time.Location)(0x1d1cea0)}}, Reason:"", Message:""}, v1.ClusterOperatorStatusCondition{Type:"Failing", Status:"False", LastTransitionTime:v1.Time{Time:time.Time{wall:0x0, ext:63689702787, loc:(*time.Location)(0x1d1cea0)}}, Reason:"", Message:""}}, Versions:[]v1.OperandVersion{v1.OperandVersion{Name:"operator", Version:"0.0.1-snapshot"}}, RelatedObjects:[]v1.ObjectReference(nil), Extension:runtime.RawExtension{Raw:[]uint8(nil), Object:runtime.Object(nil)}}}' due to: 'no kind is registered for the type v1.ClusterOperator in scheme "github.com/openshift/cluster-api/pkg/client/clientset_generated/clientset/scheme/register.go:31"'. Will not report event: 'Warning' 'Status failing' 'timed out waiting for the condition'
E0401 08:13:09.781746       1 sync.go:36] Error syncing cluster api controller: timed out waiting for the condition
```

### CODE REVIEW

File: cmd/machine-api-operator/main.go [hunk 1]
```diff
@@ -24,6 +27,8 @@ var (
 )
 
 func init() {
+	osconfigv1.AddToScheme(scheme.Scheme)
```
Comment (bison) [inline cmd/machine-api-operator/main.go]: Outside of tests where it's a quick-fix, I'm not sure `init()` is the right place for this. Shouldn't it happen in `operator.New()` or something?

Comment (enxebre) [inline cmd/machine-api-operator/main.go]: this won't be picked here https://github.com/openshift/machine-api-operator/blob/master/pkg/operator/operator.go#L23
and https://github.com/openshift/machine-api-operator/blob/master/pkg/operator/operator.go#L85

Comment (ingvagabund) [reply cmd/machine-api-operator/main.go]: Danke

Comment (ingvagabund) [reply cmd/machine-api-operator/main.go]: `init()` is called only once, `operator.New` can be called multiple times. Imho, `AddToSheme` will fail if the same type is registered twice.

File: pkg/operator/operator.go [hunk 2]
```diff
@@ -33,6 +33,10 @@ const (
 	ownedManifestsDir = "owned-manifests"
 )
 
+func init() {
+	osconfigv1.AddToScheme(scheme.Scheme)
```
Comment (bison) [inline pkg/operator/operator.go]: Right package, but I still don't think this belongs in `init()`. Why not just call this in `New()`?

Comment (bison) [reply pkg/operator/operator.go]: > init() is called only once, operator.New can be called multiple times. Imho, AddToSheme will fail if the same type is registered twice.

I really don't like putting stuff in `init()` here personally. We could set up a new scheme to add these types to and pass that when creating the event recorder, right? That would solve the problem of being called more than once.

Comment (ingvagabund) [reply pkg/operator/operator.go]: Moved under `New()`.

**Pull request merged**